lib/collection/src/operations/mod.rs

```diff
@@ -38,9 +38,68 @@ pub enum FieldIndexOperations {
     DeleteIndex(String),
 }
 
-#[derive(Debug, Deserialize, Serialize, Clone)]
-#[serde(rename_all = "snake_case")]
-#[serde(untagged)]
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct WithMeta {
+    #[serde(flatten)]
+    pub operation: CollectionUpdateOperations,
+
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub metadata: Option<OperationMeta>,
+}
+
+impl WithMeta {
```

Comment:
`WithMeta` looks more like a trait name. Maybe `OperationWithMeta`? 
Also `metadata` in general might be mixed up with payload. Not sure what would be a better alternative, though